[Profiler] Include more uncategorized events in memory profile

aaronenyeshi opened the pull request:
Summary: This PR adds handling for allocations / frees which we cannot prove are for Tensors. (And thus aren't assigned an ID.) These events are still important for judging overall utilization.

Test Plan: CI and Unit tests.

Differential Revision: D45458885

Pulled By: aaronenyeshi

anupambhatnagar reviewed (approved):
LGTM

davidberard98 commented on torch/profiler/_memory_profiler.py:
```diff
@@ -97,7 +105,6 @@ class TensorKey:
-    device: torch.device
 
     def __repr__(self) -> str:
         return f"id={self.id}: {repr(self.storage):<24} ({self.device})"
```
probably need to remove self.device here?

aaronenyeshi replied:
Since we are inheriting self.device from Key, is the better idea to make a __repr__ in class Key?

davidberard98 replied:
ah I missed that, it looks fine then!

davidberard98 reviewed (approved):
quick comment about a stray `self.device`, otherwise LGTM!

davidberard98 reviewed (approved):
quick comment about a stray `self.device`, otherwise LGTM!

aaronenyeshi commented:
@pytorchbot rebase -s

aaronenyeshi commented:
@pytorchbot merge

aaronenyeshi commented:
@pytorchbot merge

aaronenyeshi committed:
[Profiler] Include more uncategorized events in memory profile (#101200)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/101200

This PR adds handling for allocations / frees which we cannot prove are for Tensors. (And thus aren't assigned an ID.) These events are still important for judging overall utilization.

Test Plan: CI and Unit tests.

Reviewed By: anupambhatnagar, davidberard98

Differential Revision: D45458885

Pulled By: aaronenyeshi

fbshipit-source-id: 97f0baf4c378acba1828816f6ee92f3e8e4cffeb
